### ISSUE TITLE
Write xtensor array directly to h5 dataset

### ISSUE THREAD

**Issue by jamesbrownnz** [BlueBrain/HighFive#595]
I would like to be able to write an xarray/xtensor/xtensor_fixed directly to a dataset.

Currently, I get the following error:
`error C2338: static_assert failed: 'supported datatype should be an arithmetic value, a std::string or a container/array'`

```c++
#include "xtensor_h5.h"
#include <vector>
#include <cstdio>

#include <highfive/H5File.hpp>
#include <xtensor/xtensor.hpp>

using namespace HighFive;
using namespace std;

int main()
{
	remove("new_file.h5");

	// we create a new hdf5 file
	File file("new_file.h5", File::ReadWrite | File::Create | File::Truncate);

	xt::xtensor<double, 2> data{ {1.0, 2.0, 3.0},
							    {2.0, 5.0, 7.0},
							    {2.0, 5.0, 7.0} };

	// let's create a dataset of native integer with the size of the vector 'data'
	DataSet dataset = file.createDataSet<double>("/dataset_one", DataSpace::From(data));

	// let's write our vector of int to the HDF5 dataset
	dataset.write(data);

	// read back
	std::vector<int> result;
	dataset.read(result);

	return 0;
}
```

Using vcpkg:
high-five: 2.3#2
xtensor: 0.24.0
xtensor-io: 0.13.0#1

**Comment by alkino** [comment]
today xtensor is only available with H5Easy it should be fix by #594

**Comment by jamesbrownnz** [comment]
Okay I'll just close the comment as it's pretty much a duplicate of #594
